tests/versioning_3_test.go

```diff
@@ -722,6 +906,52 @@ func respondCompleteWorkflow(
 	}
 }
 
+func respondScheduleNexusOperation(
+	tv *testvars.TestVars,
+	behavior enumspb.VersioningBehavior,
+	endpointName string,
+) *workflowservice.RespondWorkflowTaskCompletedRequest {
+	return &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enumspb.COMMAND_TYPE_SCHEDULE_NEXUS_OPERATION,
+				Attributes: &commandpb.Command_ScheduleNexusOperationCommandAttributes{
+					ScheduleNexusOperationCommandAttributes: &commandpb.ScheduleNexusOperationCommandAttributes{
+						Endpoint:  endpointName,
+						Service:   "service",
+						Operation: "operation",
+						Input:     mustToPayload("input"),
+					},
+				},
+			},
+		},
+		ForceCreateNewWorkflowTask: false,
+		Deployment:                 tv.Deployment(),
+		VersioningBehavior:         behavior,
+	}
+}
+
+func mustToPayload(v any) *commonpb.Payload {
+	conv := converter.GetDefaultDataConverter()
+	payload, err := conv.ToPayload(v)
+	if err != nil {
+
+		return &commonpb.Payload{}
+	}
+	return payload
+}
+
+func respondQueryTaskCompleted(
+	task *workflowservice.PollWorkflowTaskQueueResponse,
+	namespace string,
+) *workflowservice.RespondQueryTaskCompletedRequest {
+	request := &workflowservice.RespondQueryTaskCompletedRequest{
+		Namespace: namespace,
+		TaskToken: task.TaskToken,
+	}
+	return request
+}
```

Comment:
Do you need this helper? Shouldn't you just be able to return `&workflowservice.RespondQueryTaskCompletedRequest{...}` and the taskpoller fills in the two fields for you?

Reply:
Seeing the thumbs up, did you intend to inline this now?